lib/scope.js

```diff
@@ -2485,10 +2485,35 @@ module.exports = {
       if ( !elem ) { reports.addToReport(scope, { type: `error`, value: msg }); }
       expect( elem, msg ).to.exist;
 
-      scope.toDownload = filename;
-      // Should this be using `scope.tapElement()`?
-      await elem.evaluate( elem => { return elem.click(); });
-      await scope.detectDownloadComplete( scope );
+      failed_to_download = false;
+      try {
+        const binaryStr = await scope.page.evaluate(el => {
+          const url = el.getAttribute("href");
+          return new Promise(async (resolve, reject) => {
+            const response = await fetch(url, {method: "GET"});
+            const reader = new FileReader();
+            reader.readAsBinaryString(await response.blob());
+            reader.onload = () => resolve(reader.result);
+            reader.onerror = () => reject('Error occurred while reading binary string');
```

Comment:
A chance of making the error message more clear. That might also be useful to add to the report on failure when we catch the rejection. Have to check the rest of the code to see what's going on in the download flow.

```suggestion
            reader.onerror = () => reject('Error occurred while reading binary string while trying to download a file');
```

Would be great to add the name of the file that we were attempting to download.

Reply:
I could definitely add the same of the file itself to the message, I'll do that.